added the button to import students by a csv or txt file

SarahCovey opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)
<!-- Reviewable:end -->

mrniket commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed, 2 unresolved discussions (waiting on @SarahCovey)

---

*[portal/templates/portal/teach/class.html, line 171 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6MPrCtnON5Mh-DHT:-LFw6MPrCtnON5Mh-DHU:b-t3i9p9) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/class.html#L171)):*
> ```HTML
>     function showNames(){
>         names = fileReader.result;
>         for(var index = names.indexOf(","); index!=-1; index = names.indexOf(",")){
> ```

An easier/more readable way to do this could be to do `names.split(",")`  and then `names.join("'n")`. or even together: `names.split(",").join("'n")`

https://www.w3schools.com/jsref/jsref_split.asp
https://www.w3schools.com/jsref/jsref_join.asp

---

*[portal/templates/portal/teach/onboarding_students.html, line 77 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6v-D5_XqmuZD-jeD:-LFw6v-D5_XqmuZD-jeE:b-jiwkhp) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/onboarding_students.html#L77)):*
> ```HTML
> 
>     var fileReader, theText = document.getElementById("form-create-students").elements.item(1);
>     function writeStudents(){
> ```

I see this javascript is repeated, is it possible to put it in a separate file and reuse it?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

SarahCovey commented:
Review status: 1 of 3 files reviewed, 2 unresolved discussions (waiting on @mrniket and @SarahCovey)

---

*[portal/templates/portal/teach/class.html, line 171 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6MPrCtnON5Mh-DHT:-LFwNgKfCCXMSd4c6O24:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/class.html#L171)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

An easier/more readable way to do this could be to do `names.split(",")`  and then `names.join("'n")`. or even together: `names.split(",").join("'n")`

https://www.w3schools.com/jsref/jsref_split.asp
https://www.w3schools.com/jsref/jsref_join.asp
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

SarahCovey commented:
Review status: 1 of 4 files reviewed, 2 unresolved discussions (waiting on @mrniket)

---

*[portal/templates/portal/teach/onboarding_students.html, line 77 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6v-D5_XqmuZD-jeD:-LG0suyBBXwhuoArYStt:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/onboarding_students.html#L77)):*
<details><summary><i>Previously, mrniket (Niket Shah) wrote…</i></summary><blockquote>

I see this javascript is repeated, is it possible to put it in a separate file and reuse it?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Reviewed 1 of 3 files at r1, 3 of 3 files at r2.
Review status: all files reviewed, 4 unresolved discussions (waiting on @mrniket and @SarahCovey)

---

*[portal/static/portal/js/import_student_spreadsheet.js, line 5 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LGASgeX0nvCcw-t0ujp:-LGASgeX0nvCcw-t0ujq:b97frry) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/48bcf9eb84ae7529d2211028136f57a94781be33/portal/static/portal/js/import_student_spreadsheet.js#L5)):*
> ```JavaScript
>     }
> 
>     var fileReader, theText = document.getElementById("form-create-students").elements.item(1);
> ```

Could you please add a blank line between the var and the function?

---

*[portal/static/portal/js/import_student_spreadsheet.js, line 10 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LGASk7L46_v2Llf6bFZ:-LGASk7L46_v2Llf6bF_:b-bs85qc) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/48bcf9eb84ae7529d2211028136f57a94781be33/portal/static/portal/js/import_student_spreadsheet.js#L10)):*
> ```JavaScript
>         fileReader.readAsText(document.getElementById("studentNamesFileImport").files[0]);
>         fileReader.addEventListener("loadend", showNames);
>     }
> ```

Could you add a blank line between the two functions?

---

*[portal/templates/portal/teach/class.html, line 171 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6MPrCtnON5Mh-DHT:-LGATDK14MgzV_czRORj:bn7e4ln) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/class.html#L171)):*
<details><summary><i>Previously, SarahCovey wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Why are there two script tags? Could you add the document.ready() bit inside the first script tag? 

---

*[portal/templates/portal/teach/onboarding_students.html, line 77 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6v-D5_XqmuZD-jeD:-LGATQjN3hxhY4z-vpio:b3y2xov) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/onboarding_students.html#L77)):*
<details><summary><i>Previously, SarahCovey wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Same here, would it be possible to move the src inside the script tag that has document.ready()?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: all files reviewed, 4 unresolved discussions (waiting on @mrniket and @SarahCovey)

---

*[portal/templates/portal/teach/onboarding_students.html, line 77 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6v-D5_XqmuZD-jeD:-LGAU0Nv3PI9QT4sMcWO:bop2t7f) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/onboarding_students.html#L77)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Same here, would it be possible to move the src inside the script tag that has document.ready()?
</blockquote></details>

Oh, could you add a blank line at the end of this file please?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

SarahCovey commented:
Review status: 1 of 4 files reviewed, 4 unresolved discussions (waiting on @riaJha97 and @mrniket)

---

*[portal/static/portal/js/import_student_spreadsheet.js, line 5 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LGASgeX0nvCcw-t0ujp:-LGAYOuc6BoCrfh87Wiz:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/48bcf9eb84ae7529d2211028136f57a94781be33/portal/static/portal/js/import_student_spreadsheet.js#L5)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Could you please add a blank line between the var and the function?
</blockquote></details>

Done.

---

*[portal/static/portal/js/import_student_spreadsheet.js, line 10 at r2](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LGASk7L46_v2Llf6bFZ:-LGAYQN0-zIQBaCwGFRI:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/48bcf9eb84ae7529d2211028136f57a94781be33/portal/static/portal/js/import_student_spreadsheet.js#L10)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Could you add a blank line between the two functions?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed, 2 unresolved discussions (waiting on @mrniket)

---

*[portal/templates/portal/teach/class.html, line 171 at r1](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749#-LFw6MPrCtnON5Mh-DHT:-LGAbnMD-4rNca1IIHrL:boayx9l) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/047afa0f46b18f65cb94ebb978c5a9de2c1ac5c0/portal/templates/portal/teach/class.html#L171)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Why are there two script tags? Could you add the document.ready() bit inside the first script tag? 
</blockquote></details>

I know I asked you to move the function with the script tag, but does everything still work? Is this something you can run locally and check if it's okay?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/749)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Potential security risk. Needs pen testing. Will re-look at this when #650 gets picked up again.
